### PR TITLE
Makes climbable an element + railing fixes

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/55978
- https://github.com/tgstation/tgstation/pull/56518
- https://github.com/tgstation/tgstation/pull/56744
- https://github.com/tgstation/tgstation/pull/60110

## About The Pull Request
Climbing behavior is moved from /structure/ to an element. Fixes grammar and checks with climbing.

Makes railing behavior less shit.

## Why It's Good For The Game
Hardcoded var for something that exists in only like 4 objects is dumb. Railings behave really poorly on maps, especially like Glow. this should make them behave like... ya know wou'd expect railings to.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

climb da table

he do be climbin

</details>

## Changelog
:cl: RKz, necromanceranne, MrMelbert, Ghommie
code: makes climbing mechanic an element
tweak: railings can be climbed from both sides
tweak: set deployable capsule density to 0, rather than needing to be climbed into
spellcheck: climbing cases
/:cl:

### CODE REVIEW

File: code/datums/elements/climbable.dm [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	return ..()
+
+/datum/element/climbable/proc/on_examine(atom/source, mob/user, list/examine_texts)
+	SIGNAL_HANDLER
+
+	if(can_climb(source, user))
+		examine_texts += "<span class='notice'>[source] looks climbable.</span>"
+
+/datum/element/climbable/proc/can_climb(atom/source, mob/user)
+	return TRUE
```
Comment (Penwin0) [inline code/datums/elements/climbable.dm]: Can you explain what the purpose of this proc is? It doesn't seem to be overriden at any point

Comment (Tsar-Salat) [reply code/datums/elements/climbable.dm]: found a use for it 6c5dc80

File: code/datums/elements/climbable.dm [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	SIGNAL_HANDLER
+	if(user == dropped_atom && isliving(dropped_atom))
+		var/mob/living/living_target = dropped_atom
+		if(isanimal(living_target))
+			var/mob/living/simple_animal/animal = dropped_atom
+			if (!animal.dextrous)
+				return
+		if(living_target.mobility_flags & MOBILITY_MOVE)
+			INVOKE_ASYNC(src, PROC_REF(climb_structure), climbed_thing, living_target)
+			return
```
Comment (PowerfulBacon) [inline code/datums/elements/climbable.dm]: Pointless return

File: code/datums/elements/climbable.dm [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+/datum/element/climbable/Detach(datum/target, force)
+	UnregisterSignal(target, list(COMSIG_ATOM_ATTACK_HAND, COMSIG_PARENT_EXAMINE, COMSIG_MOUSEDROPPED_ONTO, COMSIG_ATOM_BUMPED))
+	return ..()
+
+/datum/element/climbable/proc/on_examine(atom/source, mob/user, list/examine_texts)
+	SIGNAL_HANDLER
+
+	if(can_climb(source, user))
+		examine_texts += "<span class='notice'>[source] looks climbable.</span>"
```
Comment (PowerfulBacon) [inline code/datums/elements/climbable.dm]: This should indicate how you climb it. `it looks like you could climb [source] by dragging yourself onto it.` or something like that

File: code/game/objects/structures/crates_lockers/crates.dm [hunk 4]
```diff
@@ -36,6 +34,13 @@
 		animation_math = new/list()
 	if(!door_anim_time == 0 && !animation_math["[door_anim_time]-[door_anim_angle]-[azimuth_angle_2]-[radius_2]-[door_hinge]"])
 		animation_list()
+		AddElement(/datum/element/climbable, climb_time = crate_climb_time * 0.5, climb_stun = 0)
+	else
+		AddElement(/datum/element/climbable, climb_time = crate_climb_time, climb_stun = 0)
```
Comment (PowerfulBacon) [inline code/game/objects/structures/crates_lockers/crates.dm]: Having this logic tied to the door animation time seems incredibly strange to me.

**Review by PowerfulBacon — DISMISSED** [review]

**Comment by Tsar-Salat** [comment]
image

DNM, encountered many runtimes in testing

File: code/datums/elements/climbable.dm (hunk 3 above)
Comment (Tsar-Salat) [reply code/datums/elements/climbable.dm]: e56a256

File: code/datums/elements/climbable.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/datums/elements/climbable.dm]: e56a256

**Comment by Tsar-Salat** [comment]
I shall make you live again!

File: code/game/objects/structures/crates_lockers/crates.dm (hunk 4 above)
Comment (Tsar-Salat) [reply code/game/objects/structures/crates_lockers/crates.dm]: Agreed. I dont like the way tg implemented it, ill just comment it out. People can just climb on an open/closed crate at the same speed.

**Comment by Tsar-Salat** [comment]
Fixed whatever that crate issue was

climable1

climable2

climable3

**Comment by Tsar-Salat** [comment]
No runtime this time.

File: code/game/objects/structures/railings.dm [hunk 5]
```diff
@@ -19,9 +21,16 @@
 /obj/structure/railing/Initialize(mapload)
 	. = ..()
 	ini_dir = dir
+	if(climbable)
+		AddElement(/datum/element/climbable)
+
+	if(density && flags_1 & ON_BORDER_1) // blocks normal movement from and to the direction it's facing.
```
Comment (itsmeow) [inline code/game/objects/structures/railings.dm]:
```suggestion
	if(density && (flags_1 & ON_BORDER_1)) // blocks normal movement from and to the direction it's facing.
```

File: code/game/objects/structures/railings.dm [hunk 6]
```diff
@@ -74,17 +83,6 @@
 		return . || mover.throwing || mover.movement_type & (FLYING | FLOATING)
 	return TRUE
 
-/obj/structure/railing/corner/CanPass()
-	..()
-	return TRUE
-
-/obj/structure/railing/proc/init_connect_loc_element()
-	var/static/list/loc_connections = list(
```
Comment (itsmeow) [inline code/game/objects/structures/railings.dm:82]: why is this de-procced?

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/railings.dm (hunk 6 above)
Comment (Tsar-Salat) [reply code/game/objects/structures/railings.dm:82]: we removed the call to it in `/obj/structure/railing/Initialize(mapload)`, and just put the code there directly

Elements/components and whatnot are usually put under /initialization, no? Doesnt make sense to me to put it in a proc if i can just put it there

File: code/game/objects/structures/railings.dm (hunk 5 above)
Comment (Tsar-Salat) [reply code/game/objects/structures/railings.dm]: done 2eb0fab

**Comment by itsmeow** [comment]
do railings work

**Comment by Tsar-Salat** [comment]
> do railings work

New Project

**Review by itsmeow — APPROVED** [review]
I don't like that variable speed isn't a thing anymore for the crate climbing but it's a decent enough tradeoff for cleaner code.

**Comment by itsmeow** [comment]
Checks failing

**Comment by Tsar-Salat** [comment]
Alright, cleared the error with deployable capsules.

I dont really see why the capsule needs to be climbed into like crates when mobs cant even be stored in them, so im just gonna set the density to FALSE like you'd expect them to be.

**Pull request merged**